pkg/server/auth/claims.go

```diff
@@ -45,16 +45,22 @@ func (c *ClaimsConfig) PrincipalFromClaims(token claimsToken) (*UserPrincipal, e
 	groups := []string{}
 
 	if v, ok := claims[groupsKey]; ok {
+		
 		gv, ok := v.([]interface{})
-		if !ok {
-			return nil, fmt.Errorf("invalid groups claim %q in response %v", groupsKey, v)
-		}
 
-		for _, v := range gv {
-			if s, ok := v.(string); !ok {
-				return nil, fmt.Errorf("invalid groups claim %q in response %v", groupsKey, v)
-			} else {
+		if ok {
+			for _, v := range gv {
+				if s, ok := v.(string); !ok {
+					return nil, fmt.Errorf("invalid groups claim %q in response %v", groupsKey, v)
+				} else {
```

Comment:
You don't need an `else` here or below, as you're returning.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
Yeah, 100% fair point, missed that :)

Reply:
I can do the same for the 2nd if, 
But I actually prefer that the condition is truthful with AND `ok && len(s) > 0`
as opposed to `!ok || len(s) == 0`.